tests/orm/implementation/test_backend.py

```
@@ -169,28 +171,44 @@ def test_delete_nodes_and_connections(self):
 
-    with pytest.warns(ResourceWarning, match='StorageBackend was not closed explicitly'):
+    with caplog.at_level(logging.WARNING, logger=storage_backend_module.LOGGER.name):
         backend.__del__()
 
+    # Note: we do an in assert because it might be that a backend instance of a
+    # previous tes got deleted during the log capture
+    (
+        storage_backend_module.LOGGER.name,
+        logging.WARNING,
+        f'StorageBackend was not closed explicitly: {backend!r}',
+    ) in caplog.record_tuples
```

Ruff (0.15.20)

[warning] Pointless comparison. Did you mean to assign a value? Otherwise, prepend `assert` or remove it.

(B015)